### PR TITLE
Vector

### PR DESCRIPTION
**Pull request by goose4429**
## About The Pull Request

WIP. TODO: 
- Second sprite
- Mag sprites
- Make small mags fit in both pistol and ammo pouches

Adds .45 ACP KRISS Vector to the game. It is a versatile weapon, with good CQC performance.
- 25-round mag or 50-round drum mag.
- Drum mags are limited to 4 roundstart, and have to be bought in cargo.
- Drum mags are slow to reload, make you somewhat slower, and significantly hinder unwielded (akimbo) firing.
- 45 ACP has no pen, higher damage, damage falloff.
- Weapon itself has high ROF (without lagging the server, 0.15) with strong stats when wielded.
- Unwielded/akimbo is in line with MP19, with less DPS.
- Damage falloff make it much less useful on longer distances, along with accuracy penalty.

## Why It's Good For The Game

Mix up the lack of SMGs. Good versatile weapon which is actually viable.

Showcase:

https://user-images.githubusercontent.com/42085626/169531206-b9257ecb-7660-4b1b-9823-396e9f90c498.mp4

https://user-images.githubusercontent.com/42085626/169532781-67dbadcc-75fe-4040-8a64-14f62417b688.mp4

https://user-images.githubusercontent.com/42085626/169532797-e01bfa0a-991e-4569-b524-5ff5247e9528.mp4

MP19 for comparison:

It has MUCH higher accuracy on long distances, and scatter is only a problem when bursting (check beginning of the video closely).

https://user-images.githubusercontent.com/42085626/169525009-6fc3082a-d677-4f01-aa7f-625e25775b5a.mp4

## FAQ

Q: Low ROF?
A: Can't make it higher without breaking the game. Ask MJP how many pings did he get when his minigun at 10RPS was merged.

Q: Only 4 drums?
A: 4 drums are free. Buy more at cargo.

Q: Underslug?
A: Only works wielded, as always. But can remove no problem, if at least some arguments for that are given.

## Changelog
:cl:
add: new SMG-45
add: Drum mags for SMG-45
/:cl:

### CODE REVIEW

**Comment by Otterwatch** [comment]
Can you provide a video of it working in a private server?

**Comment by Araulius** [comment]
having req specific mags for standard guns is a neat idea that I want to see more of. would give increased relevance to boxes of bullets. would love to see them for things like the t-12 and what not.

**Comment by goose4429** [comment]
> Can you provide a video of it working in a private server?

Done

**Comment by FosterGrime-Heymynameslouie** [comment]
im not an art expert or anything but its sprite looks rather bland/color mute when compared to the other guns in tgmc
maybe add a bit more shading/color onto it?

File: code/modules/projectiles/guns/smgs.dm [hunk 1]
```diff
@@ -353,3 +353,83 @@
 	scatter_unwielded = 4
 	aim_slowdown = 0.15
 	wield_delay = 0.2 SECONDS
+
+
+//-------------------------------------------------------
+//Vector, based on KRISS vector 45ACP.
+/obj/item/weapon/gun/smg/vector
```
Comment (Pariah919) [inline code/modules/projectiles/guns/smgs.dm:360]:
```suggestion
/obj/item/weapon/gun/smg/vector
```
We (quite literally) have a standardized naming form for this. Probably heavy_smg or something.

should be standard_heavysmg or whatever hits your fancy as a term

File: code/modules/projectiles/guns/smgs.dm [hunk 2]
```diff
@@ -353,3 +353,83 @@
...
+	name = "\improper SMG-45 submachinegun"
```
Comment (Pariah919) [inline code/modules/projectiles/guns/smgs.dm:361]:
```suggestion
	name = "\improper SMG-45 submachinegun"
```
Find another classification term that isn't submachinegun, if you can.

File: code/modules/projectiles/guns/smgs.dm [hunk 3]
```diff
@@ -353,3 +353,83 @@
...
+	)
+
+	flags_gun_features = GUN_CAN_POINTBLANK|GUN_AMMO_COUNTER
+	gun_firemode_list = list(GUN_FIREMODE_SEMIAUTO, GUN_FIREMODE_BURSTFIRE, GUN_FIREMODE_AUTOMATIC, GUN_FIREMODE_AUTOBURST)
+	attachable_offset = list("muzzle_x" = 33, "muzzle_y" = 16, "rail_x" = 22, "rail_y" = 21, "under_x" = 25, "under_y" = 14, "stock_x" = 24, "stock_y" = 10)
+	actions_types = list(/datum/action/item_action/aim_mode)
+
+	fire_delay = 0.15 SECONDS
+	damage_mult = 1
+	damage_falloff_mult = 1.5 //shorter effective range
```
Comment (Pariah919) [inline code/modules/projectiles/guns/smgs.dm:414]:
```suggestion
	damage_falloff_mult = 1.5 //shorter effective range
```
You have a unique bullet, put it there.

File: code/modules/projectiles/guns/smgs.dm [hunk 4]
```diff
@@ -353,3 +353,83 @@
...
+	recoil = -5
+	recoil_unwielded = -2
+	wield_delay = 0.3 SECONDS
+
+	akimbo_additional_delay = 0.33
+	aim_fire_delay = 0.1 SECONDS
+	aim_speed_modifier = 1
+	aim_slowdown = 1 //Can aim without slowdown
```
Comment (Pariah919) [inline code/modules/projectiles/guns/smgs.dm:422]:
```suggestion
	aim_slowdown = 1 //Can aim without slowdown
```
Comment is wrong, this gives it slowdown.

File: code/modules/projectiles/magazines/smgs.dm [hunk 5]
```diff
@@ -117,3 +117,28 @@
...
+//Vector, based on KRISS vector 45ACP.
+/obj/item/ammo_magazine/smg/vector
+	name = "\improper SMG-45 magazine (.45ACP)"
+	desc = "A .45ACP caliber magazine for the SMG-45."
+	default_ammo = /datum/ammo/bullet/smg/acp
+	caliber = CALIBER_45ACP
+	icon_state = "uzi_ext"
+	max_rounds = 25 // MagEx 25+ mag
+
+/obj/item/ammo_magazine/smg/vector/drum
```
Comment (Pariah919) [inline code/modules/projectiles/magazines/smgs.dm:131]:
```suggestion
/obj/item/ammo_magazine/smg/vector/drum
```
Eh, we got rid of extended magazines for good reason, I don't see why we need these.

**Review by Pariah919 — CHANGES_REQUESTED** [review]
I don't personally see the niche, but its a neat gun, and we could use another SMG, I suppose.

**Comment by isotope436** [comment]
The only two concerns I have is the burst rate and underslung. The burst rate is on pare if not BETTER then the AR11 which at the moment is the best AR we have souly because it can burst as hard. also id like to see what this does to the other tiers as using just the warrior doesn't give much of a basis since 90% of the time you will be fighting other xenos (t1's in specific) and as much as I know balance is annoying for people running it against every xeno or testing it again ai xenos is always a good thing to do. 

As for the underslung that straight up makes this on pare with most shotguns in terms of straight up damage potential. half a mag of this into a runner (stick mag) with one maybe two incin nades shot out the underslung and you have killed confirmed (now this isn't as bad as when we had impacts but if I can put adhesive nades into this oh yeah its gonna murder t1s EASILY) which with how you can just middle click fire underslung now there is no time in-between. 

I'm also trying to consider this from both factions as well this is absolutely funny for marines (outside of the tan color to marines usual gray/black which only slightly angers me :P) and GOD knows we could use a new smg since there hasn't been one in the 2 years iv played, xenos are 100% gonna complain about this and honestly I can see why with it having auto burst with that firerate and the underslung this thing could run even the shotguns for their money.

In the long term however those are only balance issues and my nitpick on the color :P I for one support the addition of another smg to the mix and actually had plans to add my old modules smg with a color rework to fill the void. (might still add it as a specialist or req item though :P) so hey go ham with it :D (also I'd keep in mind the idea of only 4 drum mags and if ammo packs will be available for it since a lot of new guns tend not to have ammo packs for them :P)

File: code/modules/projectiles/magazines/smgs.dm (hunk 5 above)
Comment (isotope436) [reply code/modules/projectiles/magazines/smgs.dm:131]: there actually just repurposed ppsh mags by the looks of it so they are already in game dont know how that works for code though and honestly the reason for why drum mags got removed was both unknown to a lot of people and not really much of an issue anymore given how way xenos operate now (and with every smg in import basically being drum magged least we forget about the smg 25s 60 round mags :P)

**Comment by isotope436** [comment]
also i didnt notice till now but those sprites need some work the forward facing sprite looks half complete and theres a stock on both sides. kuros probably gonna bully you about it :P

**Review by Kuro020 — CHANGES_REQUESTED** [review]
image
guh?

ok first off, why are there two vector sprites? leave the black one in and drop the other, the desert colored vector is gonna take way more work to fix.

the black vector is too big, its 64 pixels long for a smg, not only that but it has about 10+ different shades of color for some reason, stick to the tgmc gun pallete, fix the shading and reduce this gun to about the size of the p90.

File: code/modules/projectiles/guns/smgs.dm (hunk 3 above)
Comment (goose4429) [reply code/modules/projectiles/guns/smgs.dm:414]: very true

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 6]
```diff
@@ -34,6 +34,9 @@
 			/obj/item/ammo_magazine/smg/standard_smg = -1,
 			/obj/item/weapon/gun/smg/standard_machinepistol = -1,
 			/obj/item/ammo_magazine/smg/standard_machinepistol = -1,
+			/obj/item/weapon/gun/smg/standard_stormsmg = -1,
+			/obj/item/ammo_magazine/smg/standard_stormsmg = -1,
+			/obj/item/ammo_magazine/smg/standard_stormsmg/drum = 4,
```
Comment (BraveMole) [inline code/game/objects/machinery/vending/marine_vending.dm:39]: either both mags are balanced and free, or only one exist. Not that

File: code/modules/projectiles/guns/smgs.dm [hunk 7]
```diff
@@ -353,3 +353,82 @@
...
+	force = 20
+	type_of_casings = null
+	//default_ammo_type = /obj/item/ammo_magazine/pistol/standard_heavypistol
+	//allowed_ammo_types = list(/obj/item/ammo_magazine/pistol/standard_heavypistol)
+	default_ammo_type = /obj/item/ammo_magazine/smg/standard_stormsmg
+	allowed_ammo_types = list(
+		/obj/item/ammo_magazine/smg/standard_stormsmg,
+		/obj/item/ammo_magazine/smg/standard_stormsmg/drum,
+	)
+	attachable_allowed = list(
```
Comment (BraveMole) [inline code/modules/projectiles/guns/smgs.dm:379]: a SMG with that much attachable is not an smg...

like, scopes ? bipod ???? flamer and grenade launcher ????? 
It should have the same attachables than a smg90

**Review by BraveMole — CHANGES_REQUESTED** [review]
what is the niche compared to a smg90 for example ?

File: code/game/objects/machinery/vending/marine_vending.dm (hunk 6 above)
Comment (goose4429) [reply code/game/objects/machinery/vending/marine_vending.dm:39]: Why?

Comment (isotope436) [reply code/game/objects/machinery/vending/marine_vending.dm:39]: because it will get slotted into imports

**Pull request closed without merging**